### DISCUSSION TITLE
How to use the data of optitrack in the controller?

### DISCUSSION THREAD

**Discussion by Carmol1** [USC-ACTLab/crazyswarm#655] [Q&A]
Hi.

How can i access to the data obtained of the optitrack cameras to use them in a new control law or in one of the existing in crazyflie-firmware ?.

I would appreciate your help.

**Comment by whoenig** [comment]
If you are looking for implementing your own on-board controller in C, you can get help at discussions.bitcraze.io. If you are looking in implementing an off-board controller in ROS, we might be able to help you here, but you would need to specify what your exact requirements are. In general, off-board controllers are not possible for the attitude part, because the communication latency is too high.

**Reply by Carmol1** [reply]
Hello.

I appreciate your answer, if I am looking to implement my own control law for both position and attitude using c or python, on the other hand, as you mention I have already published in that forum and I am waiting for an answer, regarding the off- board controller I accept your help, since it could be very useful to me, my exact requirements are that a group of crazyflies follow a trajectory, but at the moment I am only interested in controlling one, as control inputs I obtain the thrust force and the torques in x,y,z.

Greetings.

**Reply by jpreiss** [reply]
If your controller produces torque commands, latency will make it difficult/impossible to run it offboard.

Usually offboard controllers are producing something high-level like a desired velocity or acceleration vector.

**Reply by Carmol1** [reply]
Hello.

Thank you for answering, very well then to solve what you mention, I would use the following equation
![SmartSelect_20220601-205237_WhatsApp](https://user-images.githubusercontent.com/104467412/171550634-3af6453c-c79a-4906-b457-34a984acb27a.jpg)
to obtain the vector of angular velocities.

Greetings
